File: src/main/java/com/cayena/backend/entities/Product.java
package com.cayena.backend.entities;

import javax.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDate;
@Entity
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long productId;
    private String name;
    private Integer quantityInStock;
    private BigDecimal unitPrice;
    @ManyToOne
    @JoinColumn(name = "supplier_id")
    private Suplier suplierId;
    private LocalDate dateOfCreation;
    private LocalDate dateOfTheLastUpdate;
}

File: backend-kotlin/src/main/kotlin/com/cayena/backendkotlin/dominio/Product.kt
package com.cayena.backendkotlin.dominio

import java.math.BigDecimal
import java.time.LocalDateTime
import javax.persistence.*


@Entity
data class Product(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val productId: Long? = null,
    val name: String,
    val quantityInStock: Int,
    val unitPrice: BigDecimal,
    val dateOfCreation: LocalDateTime? = LocalDateTime.now(),
    val dateOfThelastUpdate: LocalDateTime? = LocalDateTime.now(),

    @ManyToOne
    @JoinColumn(name = "supplier_id")
    val supplier: Supplier
)